### PR TITLE
[TTAHUB-1826] Added permissions for edit and create on training report landing

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

Added permissions for edit and create on training report landing.

## How to test

- Create a user with only read only permissions for training reports. You should not have the ability to create or edit a event or session.
- Create a user with read write permission for training reports. You should have the ability to create and edit event and session reports.
- The view option should be available for both paths.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1826

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/pages/TrainingReports/components/EventCard.js [hunk 1]
```diff
@@ -25,16 +33,29 @@ function EventCard({
...
+    }
+
+    // View event.
     menuItems.push({
-      label: 'Create session',
-      onClick: () => {
-        history.push(`/training-report/${event.id}/session/new/`);
-      },
-    });
-    menuItems.push({
```
Comment (thewatermethod) [inline frontend/src/pages/TrainingReports/components/EventCard.js:34]: You should be able to view completed events

**Comment by AdamAdHocTeam** [comment]
> It'd be nice if this PR also handled the collaborator permission.

Yea I can add that, would it just be the ability to edit and not create?

File: frontend/src/pages/TrainingReports/components/EventCard.js (hunk 1 above)
Comment (AdamAdHocTeam) [reply frontend/src/pages/TrainingReports/components/EventCard.js:34]: Ok so should always add the view item?

Comment (thewatermethod) [reply frontend/src/pages/TrainingReports/components/EventCard.js:34]: Yeah looks like it, might be worth reviewing the mockups to double check my thinking

**Comment by thewatermethod** [comment]
> Yea I can add that, would it just be the ability to edit and not create?

@AdamAdHocTeam I'm thinking specifically of the "regional point of contact" collaborator

https://www.figma.com/file/RZqI76V9igIpWUkHB2KdUm/PD-Event-Form?node-id=278%3A46244&mode=dev

File: frontend/src/pages/TrainingReports/components/EventCard.js (hunk 1 above)
Comment (AdamAdHocTeam) [reply frontend/src/pages/TrainingReports/components/EventCard.js:34]: This is updated.

File: frontend/src/pages/TrainingReports/components/EventCard.js [hunk 2]
```diff
@@ -1,18 +1,29 @@
...
 
 function EventCard({
   event,
 }) {
+  const { user } = useContext(UserContext);
+  const hasEditPermissions = canEditOrCreateSessionReports(
+    user,
+    parseInt(event.regionId, DECIMAL_BASE),
+  );
+  const isCollaborator = event.collaboratorIds.includes(user.id);
```
Comment (thewatermethod) [inline frontend/src/pages/TrainingReports/components/EventCard.js:24]:
```suggestion
  const isCollaborator = event.pocId.includes(user.id);

```

Comment (thewatermethod) [reply frontend/src/pages/TrainingReports/components/EventCard.js:24]: I know its a little weird, but collaboratorIds is "IST Collaborators" and "pocId" is "Regional Point of Contact Collaborators"

**Comment by thewatermethod** [comment]
Tested out, everything looks good, just need to address the one thing and we are good to go with this

File: frontend/src/pages/TrainingReports/components/EventCard.js (hunk 2 above)
Comment (nvms) [reply frontend/src/pages/TrainingReports/components/EventCard.js:24]: mostly unrelated to this PR: (sorry to hijack)

I didn't realize `pocId` was migrated to an array. I think the tests are all still written to validate it as a `number`.

come to think of it, the `createEvent` handler just takes whatever you give it for the value and sticks it in the db, but this test is still passing??

https://github.com/HHS/Head-Start-TTADP/blob/f18cf62fdd84b81c107b84dac7bedd1b3a95da83/src/routes/events/handlers.test.js#L125-L145

Comment (thewatermethod) [reply frontend/src/pages/TrainingReports/components/EventCard.js:24]: I updated the services test, but I guess I missed this one! 

Looks like the only validation that is happening is whether or not pocId is truthy, unless I'm missing something somewhere?

Comment (thewatermethod) [reply frontend/src/pages/TrainingReports/components/EventCard.js]: as would I

**Review by thewatermethod — APPROVED** [review]

**Pull request merged**